source-dropbox/main.go

```diff
@@ -0,0 +1,300 @@
+package main
+
+import (
+	"cmp"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"slices"
+	"time"
+
+	"github.com/dropbox/dropbox-sdk-go-unofficial/v6/dropbox"
+	"github.com/dropbox/dropbox-sdk-go-unofficial/v6/dropbox/files"
+	"github.com/estuary/connectors/filesource"
+	"github.com/estuary/flow/go/parser"
+	pf "github.com/estuary/flow/go/protocols/flow"
+)
+
+type config struct {
+	Path        string         `json:"path"`
+	Credentials *Credentials   `json:"credentials"`
+	Parser      *parser.Config `json:"parser"`
+	MatchKeys   string         `json:"matchKeys,omitempty"`
+	Advanced    advancedConfig `json:"advanced"`
+}
+
+type advancedConfig struct {
+	AscendingKeys bool `json:"ascendingKeys,omitempty"`
+}
+
+func (c config) Validate() error {
+	requiredProperties := [][]string{
+		{"Path", c.Path},
+	}
+	for _, req := range requiredProperties {
+		if req[1] == "" {
+			return fmt.Errorf("missing '%s'", req[0])
+		}
+	}
+	return nil
+}
+
+func (c config) DiscoverRoot() string {
+	return c.Path
+}
+
+func (c config) RecommendedName() string {
+	return c.Path[1:]
```

Comment:
I think it would be more clear to use `strings.Trim` here, since I believe the intent is to return the path with the leading slash removed.

On most other filesource captures we are using `strings.Trim(c.DiscoverRoot(), "/")`.

Reply:
I agree, I'll change it to use `strings.Trim`.